components/raftstore/src/store/peer.rs

```diff
@@ -3464,6 +3477,18 @@ where
             send_msg.set_start_key(region.get_start_key().to_vec());
             send_msg.set_end_key(region.get_end_key().to_vec());
         }
+
+        if self.is_leader()
+            && msg.get_msg_type() == MessageType::MsgHeartbeat
+            && self.has_applied_to_current_term()
```

Comment:
Is there a way to send the payload only when it's necessary?

Reply:
Actually, the `CheckLeader` rpc also have such payload to sync the `safe ts` to follower, and the payload here just an optimization to sync `safe ts` more frequently (since there will be no raft heartbeat if the region is hibernated), and I think the overhead here is small.

Reply:
Hibernate is optional before, this makes it a must. Have you check how many extra bytes will be added to the message comparing a normal heartbeat?

Reply:
Do you mean the payload at the `CheckLeader` rpc is unnecessary?

By setting `applied_index` to `1234` and `safe_ts` to `u64::MAX`, there are 16 extra bytes :
```
Cost of read state, current size: 48, prev size: 32
```